src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1329,6 +1329,24 @@ def show_vm(cmd, resource_group_name, vm_name, show_details=False):
         else get_vm(cmd, resource_group_name, vm_name)
 
 
+def delete_vm(cmd, vm_name=None, resource_group_name=None, tags=None):
+    from msrestazure.tools import parse_resource_id
+    vm_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_COMPUTE).virtual_machines
+    if tags is not None:
+        vms = list_vm(cmd, None, True)
+        for vm in vms:
+            # TODO compliance btw python3 and python2 ?
+            if len(tags) == len(set(tags.items()) & set(vm.tags.items())):
```

Comment:
I have question about this line. Say tags={1, 1, 2}, vm.tags={1, 2}, condition is false.

Reply:
What is the attended behavior if the `tags` flag is present but the value is empty?
For example, the following command `az vm --delete --tags "" --yes` delete in this case all VMs.

Reply:
@sagarafr At present, the behavior of `az resource list --tag ""` is to search all the resources. 
Because when `--tag` is passed in as `""`, it will be changed into an `{}` by method `validate_tag`  [codel link](https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/commands/validators.py#L50
) in `azure-cli-core`.  Then the filter condition of tag will not be added to the rest request, so it will not search for resources by tag.

I personally think the behavior of `--tags ""` in `az vm --delete --tags "" --yes` needs to be consistent with `--tag ""` in `az resource list --tag ""`.